### PR TITLE
Remove some global variables from main.go

### PR DESCRIPTION
**Pull request by andersonvom**
This is a first step towards getting rid of the global variables currently in use (plus some minor improvements along the way). The remaining globals will take quite a bit of refactoring, so they will have to be addressed in later commits.

### CODE REVIEW

File: cmd/syncthing/main.go [hunk 1]
```diff
@@ -207,132 +207,145 @@ var (
...
+	if os.Getenv("STTRACE") != "" {
+		options.logFlags = log.Ltime | log.Ldate | log.Lmicroseconds | log.Lshortfile
+	}
 
-func parseCommandLineOptions() {
 	if runtime.GOOS == "windows" {
-		// On Windows, we use a log file by default. Setting the -logfile flag
-		// to "-" disables this behavior.
-		flag.StringVar(&logFile, "logfile", "", "Log file name (use \"-\" for stdout)")
+		options.logFile = locations[locLogFile]
```
Comment (AudriusButkevicius) [inline cmd/syncthing/main.go]: This changes behaviour. It will always be "-" now, where as the default on windows used to be ""

Comment (AudriusButkevicius) [inline cmd/syncthing/main.go]: Ignore me. This will actually be fine.

File: cmd/syncthing/main.go [hunk 2]
```diff
@@ -207,132 +207,145 @@ var (
...
 		return
 	}
 
-	if noRestart {
-		syncthingMain()
+	if options.noRestart {
+		syncthingMain(options)
 	} else {
-		monitorMain()
+		monitorMain(options.logFile)
```
Comment (AudriusButkevicius) [inline cmd/syncthing/main.go]: less pass the options, as I bet there will be soon something we will want from there.

File: cmd/syncthing/main.go (hunk 1 above)
Comment (andersonvom) [inline cmd/syncthing/main.go]: It seemed to me the previous behavior was:

* if it's windows, defaults to blank;
* if it's not windows, defaults to "-";
* inside `main()`, if blank change it to `locations[locLogFile]`;

I think the behavior is the same, just one step shorter:
* defaults to "-"
* if it's windows, defaults to `locations[locLogFile]`

or did I miss something?

File: cmd/syncthing/main.go (hunk 2 above)
Comment (andersonvom) [inline cmd/syncthing/main.go]: Yeah, I was on the fence for that one. I guess we can pass the whole thing.

**Comment by calmh** [comment]
Neat!

**Pull request merged**